src/APICore.ts

```diff
@@ -211,7 +212,8 @@ export default class API {
     }
 
     private async request<T>(route: string, init: RequestInit, responseBodyFormat?: ResponseBodyFormat): Promise<T> {
-        const fetcher = createFetcher(this.getUrlFromRoute(route), init, isTooManyRequests);
+        const enrichedOptions = handleRequest(init, this.config.requestHandlers);
+        const fetcher = createFetcher(this.getUrlFromRoute(route), enrichedOptions, isTooManyRequests);
```

Comment:
I would change the naming to stay consistent with the parameter

```suggestion
        const enrichedRequestInit = handleRequest(init, this.config.requestHandlers);
        const fetcher = createFetcher(this.getUrlFromRoute(route), enrichedRequestInit, isTooManyRequests);
```